### PR TITLE
Support REST in Throw error Job command

### LINKED ISSUE #21412: Support the Throw error API over REST

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

Rest support for Job throw error command

## Related issues

closes #21412

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ThrowErrorCommandImpl.java [hunk 1]
```diff
@@ -30,69 +33,127 @@
...
-    builder = ThrowErrorRequest.newBuilder();
-    builder.setJobKey(key);
+    grpcRequestObjectBuilder = ThrowErrorRequest.newBuilder();
+    grpcRequestObjectBuilder.setJobKey(key);
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+    httpRequestObject = new JobErrorRequest();
+    useRest = preferRestOverGrpc;
+    jobKey = key;
+    this.jsonMapper = jsonMapper;
```
Comment (koevskinikola) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ThrowErrorCommandImpl.java:71]: 🔧 Same as with the `FailJobCommandImpl`, why do we need to have a second property for the ` jsonMapper` when the base class already provides it through the `objectMapper` property?

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Nice job @nicpuppa, I only have a minor clarification before we can merge this PR. 🚀

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ThrowErrorCommandImpl.java (hunk 1 above)
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ThrowErrorCommandImpl.java:71]: Good catch 👍

**Review by koevskinikola — APPROVED** [review]
Awesome! Looks great now @nicpuppa 🚀

**Pull request merged**